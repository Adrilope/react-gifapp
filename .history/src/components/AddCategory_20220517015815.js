import React from 'react'

export const AddCategory = () => {
  return (
    <>
    <h2></h2>AddCategory</>
  )
}
