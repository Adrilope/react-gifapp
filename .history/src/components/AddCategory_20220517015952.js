import React from 'react'

export const AddCategory = () => {
    return (
        <>
            <h3>Add Category</h2>
        </>
    )
}
