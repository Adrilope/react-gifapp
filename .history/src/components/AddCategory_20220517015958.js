import React from 'react'

export const AddCategory = () => {
    return (
        <>
            <h3>Add Category</h3>
            <input
        </>
    )
}
