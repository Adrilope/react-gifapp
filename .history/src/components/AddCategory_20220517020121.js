import React from 'react'

export const AddCategory = () => {

    co

    return (
        <>
            <input 
                type="text"

            />
        </>
    )
}
