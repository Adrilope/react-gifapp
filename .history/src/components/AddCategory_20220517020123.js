import React from 'react'

export const AddCategory = () => {

    const 

    return (
        <>
            <input 
                type="text"

            />
        </>
    )
}
