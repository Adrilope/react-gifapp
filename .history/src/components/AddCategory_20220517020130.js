import React from 'react'

export const AddCategory = () => {

    useS
    return (
        <>
            <input 
                type="text"

            />
        </>
    )
}
