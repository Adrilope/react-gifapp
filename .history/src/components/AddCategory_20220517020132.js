import React from 'react'

export const AddCategory = () => {

    const [first, setfirst] = useState(second)
    return (
        <>
            <input 
                type="text"

            />
        </>
    )
}
