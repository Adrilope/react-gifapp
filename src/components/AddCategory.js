import React, {useState} from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setinputValue] = useState('')

    const handleInputChange = (e) => {
        setinputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault()  

        if (inputValue.trim().length > 2) {
            onNewCategory(inputValue.trim())
            setinputValue('')
        }
    }


    return (
        <form onSubmit={ handleSubmit }>
            <p>{inputValue}</p>
            <input 
                placeholder='Search gifs'
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}


AddCategory.propTypes = {
    onNewCategory : PropTypes.func.isRequired
}
