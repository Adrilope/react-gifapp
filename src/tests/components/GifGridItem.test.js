import '@testing-library/jest-dom'
import { shallow } from "enzyme"
import { GifGridItem } from '../../components/GifGridItem'


describe('Tests in GifGridItem component', () => {

    const title = 'Un titulo'
    const url = 'https://localhost/algo.jpg'

    let wrapper = shallow(<GifGridItem title={title} url={url}/>)


    test('must render <GifGridItem /> with props', () => {
        expect(wrapper).toMatchSnapshot()
    })


    test('must have a p with title', () => { 
        const p = wrapper.find('p')

        expect(p.text().trim()).toBe(title)
    })


    test('should have the image with the props value of the alt & url', () => { 
        const img = wrapper.find('img')

        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    })

    
    test('should have animate__fadeInUp', () => { 
        const div = wrapper.find('div')
        const className = div.prop('className')
        
        expect(className.includes('animate__fadeInUp')).toBe(true)
    })
})