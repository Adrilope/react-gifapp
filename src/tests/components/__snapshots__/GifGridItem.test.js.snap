// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifGridItem component must render <GifGridItem /> with props 1`] = `
<div
  className="card animate__animated animate__fadeInUp"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
